packages/tests/server/middlewares.test.ts

```diff
@@ -319,6 +319,42 @@ test('standalone middlewares that define the ctx/input they require and can be u
   tHumanWithMeta.procedure.use(shamefullyLogIfProcedureIsNotCoolMiddleware);
 });
 
+test('Fix #4947: standalone middlewares -- inputs are merged properly when using multiple standalone middlewares', () => {
```

Comment:
Can you move this here 

https://github.com/trpc/trpc/tree/main/packages/tests/server/regression